lib/puppet/provider/package/dpkg.rb

```diff
@@ -81,6 +81,11 @@ def install
     unless file = @resource[:source]
       raise ArgumentError, _("You cannot install dpkg packages without a source")
     end
+
+    if :held == @resource[:ensure] && @resource[:hold]
```

Comment:
Is it always symbol `:held` or can it be the string `'held'`?

Reply:
Check was moved to package type, it is symbol there.